src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -1,164 +1,581 @@
 
-            return MakeIsDeclarationPattern(loweredPattern.Syntax, loweredInput, loweredPattern.VariableAccess, requiresNullTest: loweredInput.Type.CanContainNull());
-        }
+                // then process all of the individual decisions
+                foreach (var decision in decisions)
+                {
+                    if (LowerDecision(decision))
+                    {
+                        break;
+                    }
+                }
 
-        /// <summary>
-        /// Is the test, produced as a result of a pattern-matching operation, always true?
-        /// Knowing that enables us to construct slightly more efficient code.
-        /// </summary>
-        private bool IsIrrefutablePatternTest(BoundExpression test)
-        {
-            while (true)
-            {
-                switch (test.Kind)
+                if (_sideEffectBuilder.Count != 0)
                 {
-                    case BoundKind.Literal:
-                        {
-                            var value = ((BoundLiteral)test).ConstantValue;
-                            return value.IsBoolean && value.BooleanValue;
-                        }
-                    case BoundKind.Sequence:
-                        test = ((BoundSequence)test).Value;
-                        continue;
-                    default:
-                        return false;
+                    _conjunctBuilder.Add(_factory.Sequence(ImmutableArray<LocalSymbol>.Empty, _sideEffectBuilder.ToImmutable(), _factory.Literal(true)));
+                    _sideEffectBuilder.Clear();
+                }
+                BoundExpression result = null;
+                foreach (var conjunct in _conjunctBuilder)
+                {
+                    result = (result == null) ? conjunct : _factory.LogicalAnd(result, conjunct);
+                }
+
+                var bindingsBuilder = ArrayBuilder<BoundExpression>.GetInstance();
+                foreach (var (left, right) in bindings)
+                {
+                    bindingsBuilder.Add(_factory.AssignmentExpression(left, _tempAllocator.GetTemp(right)));
+                }
+
+                if (bindingsBuilder.Count > 0)
+                {
+                    var c = _factory.Sequence(ImmutableArray<LocalSymbol>.Empty, bindingsBuilder.ToImmutableAndFree(), _factory.Literal(true));
+                    result = (result == null) ? c : (BoundExpression)_factory.LogicalAnd(result, c);
                 }
+                else if (result == null)
+                {
+                    result = _factory.Literal(true);
+                }
+
+                return _factory.Sequence(_tempAllocator.AllTemps(), ImmutableArray<BoundExpression>.Empty, result);
             }
         }
 
-        private BoundExpression CompareWithConstant(BoundExpression input, BoundExpression boundConstant)
+        private BoundExpression MakeTypeTestAndAssignment(BoundExpression loweredTarget, BoundExpression loweredInput, TypeSymbol type)
```

Comment:
>MakeTypeTestAndAssignment [](start = 32, length = 25)

It looks like the function is not used. Please remove, I did not review its body.

Reply:
Resolved before or in of [https://github.com/dotnet/roslyn/pull/23357](https://github.com/dotnet/roslyn/pull/23357)

---
In reply to: [156562225](https://github.com/dotnet/roslyn/pull/23209#discussion_r156562225) [](ancestors = 156562225)